### PR TITLE
Migrate applicant question tests to use TestContext

### PR DESCRIPTION
**Pull request by nbeloglazov**
### Description

See #3323 for context.

### Checklist

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: browser-test/src/support/index.ts [hunk 1]
```diff
@@ -116,6 +118,7 @@ export interface TestContext {
  *   const ctx = createTestContext()
  *
  *   it('should do foo', async () => {
+  const {page, applicantQuestions} = ctx
```
Comment (jeanniefu) [inline browser-test/src/support/index.ts:121]: nit: fix formatting

File: browser-test/src/checkbox.test.ts [hunk 2]
```diff
@@ -1,40 +1,23 @@
...
-  afterEach(async () => {
-    await resetSession(pageObject)
-  })
+  const ctx = createTestContext(/* clearDb= */ false)
 
   describe('single checkbox question', () => {
-    let applicantQuestions: ApplicantQuestions
     const programName = 'test program for single checkbox'
 
     beforeAll(async () => {
```
Comment (jeanniefu) [inline browser-test/src/checkbox.test.ts:17]: not-blocking, and no action needed. just for discussion - In theory, we should be able to just change this to beforeEach for all the test files and set clearDB=true and that should work right? Do you know if there's a large performance hit?

**Review by jeanniefu — APPROVED** [review]
Thank you!

File: browser-test/src/checkbox.test.ts (hunk 2 above)
Comment (nbeloglazov) [reply browser-test/src/checkbox.test.ts:17]: Yes, replacing `beforeAll` with `beforeEach` will allow us to clean DB between tests. But as you said it does introduce performance hit and pretty nasty :/ 

I just wrote down some thoughts about that here: https://github.com/civiform/civiform/issues/2891#issuecomment-1235813976

File: browser-test/src/support/index.ts (hunk 1 above)
Comment (nbeloglazov) [reply browser-test/src/support/index.ts]: Thanks. Done.

**Pull request merged**